### PR TITLE
[ant-276] copy files using rsync

### PR DESCRIPTION
**Pull request by hazelbit**
## Motivation

Quick PR to copy the project using [rsync](https://linux.die.net/man/1/rsync) to remote macOS devices to improve resilience.  Unlike #6286 we're not attempting to skip any files or folders (as some tools might assume the existence of those). Note that compression (`-z`) led to a walltime penalty in our scenarios (see [ticket](https://jira.unity3d.com/browse/ANT-276) for more details) so it is not enabled. The ticket also contains a quick summary of why the specific flags are used. The goal of this PR is to increase resilience in case of brief network hiccups (we still want to fail the pipeline if network issues persist).

### CODE REVIEW

File: .yamato/config/__shared.metafile [hunk 1]
```diff
@@ -356,7 +356,7 @@ platforms:
           - command: curl -s https://artifactory.prd.it.unity3d.com/artifactory/unity-tools-local/utr-standalone/utr --output utr
             retries: 2
           - chmod +x utr
-          - command: scp -i ~/.ssh/id_rsa_macmini -r $YAMATO_SOURCE_DIR bokken@$BOKKEN_DEVICE_IP:~/Graphics
```
Comment (theo-pnv) [inline .yamato/config/__shared.metafile:359]: So it's implied that rsync will copy everything over inside a Graphics folder, on the destination? in other words, no need to explicitly have `~/Graphics` anymore?
I guess the test results will confirm :)

**Review by theo-pnv — APPROVED** [review]
LGTM, let's wait for the tests results.

File: .yamato/config/__shared.metafile (hunk 1 above)
Comment (hazelbit) [reply .yamato/config/__shared.metafile:359]: resolving :: [slack](https://unity.slack.com/archives/D02HZ4S7STT/p1638194181002700) 🙂

**Comment by hazelbit** [comment]
The pipeline `URP_Foundation on OSX_Metal_playmode_mono_Linear on version trunk` has the same test failures as on `master`, visible in the respective test reports ([branch](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/1a725439-fa8c-4bd7-ab3d-b889a490e054%2Flogs%2FTestProjects%2FUniversalGraphicsTest_Foundation%2Ftest-results/TestReport.html) vs [master](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/a051042c-cf60-4093-8cf9-f17873b85955%2Flogs%2FTestProjects%2FUniversalGraphicsTest_Foundation%2Ftest-results/TestReport.html)), which are not related to the changes in this PR. 🙂

**Review by theo-pnv — APPROVED** [review]

**Pull request merged**